### PR TITLE
gateway proxy http client now use ProxySelectorRoutePlanner

### PR DESCRIPTION
**Pull request by boly38**
this allow end-users to set proxy using system properties 
example: 
 -Dhttps.proxyHost=proxy -Dhttps.proxyPort=3128

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@boly38__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

**Comment by boly38** [comment]
related stackoverflow question  http://stackoverflow.com/questions/30460523/documentdb-java-sdk-behind-a-proxy

**Comment by ryancrawcour** [comment]
thanks, we'll take a look at this shortly.

File: src/com/microsoft/azure/documentdb/GatewayProxy.java [hunk 1]
```diff
@@ -20,6 +21,9 @@
 
 import org.apache.http.Header;
 import org.apache.http.HttpEntity;
+import org.apache.http.HttpException;
+import org.apache.http.HttpHost;
+import org.apache.http.HttpRequest;
```
Comment (shipunyc) [inline src/com/microsoft/azure/documentdb/GatewayProxy.java]: Please remove these headers, which are not used. Ecplise gives warnings.

File: src/com/microsoft/azure/documentdb/GatewayProxy.java [hunk 2]
```diff
@@ -28,11 +32,16 @@
 import org.apache.http.client.methods.HttpPost;
 import org.apache.http.client.methods.HttpPut;
 import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.conn.routing.HttpRoute;
+import org.apache.http.conn.routing.HttpRoutePlanner;
+import org.apache.http.conn.scheme.SchemeRegistry;
```
Comment (shipunyc) [inline src/com/microsoft/azure/documentdb/GatewayProxy.java]: Remove un-used headers.

File: src/com/microsoft/azure/documentdb/GatewayProxy.java [hunk 3]
```diff
@@ -154,7 +163,13 @@ private HttpClient getHttpClient(boolean isForMedia) {
...
     private HttpClient createHttpClient(boolean isForMedia) {
-        HttpClient httpClient = new DefaultHttpClient(this.connectionManager);
+        ProxySelectorRoutePlanner ps = new ProxySelectorRoutePlanner(
+                SchemeRegistryFactory.createDefault(), ProxySelector.getDefault());
+
+        DefaultHttpClient defaultHttpClient = new DefaultHttpClient(this.connectionManager);
+        defaultHttpClient.setRoutePlanner(ps);
+        
+        HttpClient httpClient = defaultHttpClient;
         HttpParams httpParams = httpClient.getParams();
```
Comment (shipunyc) [inline src/com/microsoft/azure/documentdb/GatewayProxy.java]: Remove this line "HttpClient httpClient = defaultHttpClient;". And change the next line to be "HttpParams httpParams = defaultHttpClient.getParams();"

**Comment by boly38** [comment]
Hi Shipunyc
thanks for your feedback
I will remove unused imports (sorry for that)
But for the http client I just need to change (l157)

    HttpClient httpClient = defaultHttpClient;

by

    httpClient = defaultHttpClient;

as httpClient is a field of the class!
I will make the changes, don't hesitate to tell me If you're green with that ;)

**Comment by boly38** [comment]
Important notice: I cant test without proxy (direct connection), so I hope that some IT will test that for me.

**Comment by shipunyc** [comment]
Hi Boly, thanks for your change. I will merge.

**Pull request merged**